modules/ossm-enabling-sidecar-injection.adoc

```diff
@@ -0,0 +1,261 @@
+
+. Verify that workloads already running in the desired namespace show `1/1` containers as `READY`. This confirms that the pods are running without sidecars.
++
+[source,terminal]
+----
+$ oc get pods -n bookinfo
+----
++
+You should see output similar to the following example:
++
+.Example output
+[source,terminal]
+----
+NAME                             READY   STATUS    RESTARTS   AGE
+details-v1-65cfcf56f9-gm6v7      1/1     Running   0          4m55s
+productpage-v1-d5789fdfb-8x6bk   1/1     Running   0          4m53s
+ratings-v1-7c9bd4b87f-6v7hg      1/1     Running   0          4m55s
+reviews-v1-6584ddcf65-6wqtw      1/1     Running   0          4m54s
+reviews-v2-6f85cb9b7c-w9l8s      1/1     Running   0          4m54s
+reviews-v3-6f5b775685-mg5n6      1/1     Running   0          4m54s
+----
+
+. Run the following command at the CLI to apply the injection label to the `bookinfo` namespace:
++
+[source,terminal]
+----
+$ oc label namespace bookinfo istio-injection=enabled
+namespace/bookinfo labeled
+----
+
+. Redeploy existing workloads in the `bookinfo` namespace to enable sidecar injection. Use the following command to perform a rolling update of all workloads:
++
+[source,terminal]
+----
+$ oc -n bookinfo rollout restart deployments
+----
+
+.Verification
+
+. Verify the rollout by checking that the new pods display `2/2` containers as `READY`, confirming successful sidecar injection.
++
+[source,terminal]
+----
+$ oc get pods -n bookinfo
+----
++
+You should see output similar to the following example:
++
+.Example output
+[source,terminal]
+----
+NAME                              READY   STATUS    RESTARTS   AGE
+details-v1-7745f84ff-bpf8f        2/2     Running   0          55s
+productpage-v1-54f48db985-gd5q9   2/2     Running   0          55s
+ratings-v1-5d645c985f-xsw7p       2/2     Running   0          55s
+reviews-v1-bd5f54b8c-zns4v        2/2     Running   0          55s
+reviews-v2-5d7b9dbf97-wbpjr       2/2     Running   0          55s
+reviews-v3-5fccc48c8c-bjktn       2/2     Running   0          55sz
+----
+
+[id="ossm-enabling-sidecar-injection-exclude-workload-from-mesh_{context}"]
+== Exclude a workload from the mesh
+
+You can exclude specific workloads from sidecar injection within a namespace where injection is enabled for all workloads.
+
+[NOTE]
+====
+This example is for demonstration purposes only. The bookinfo application requires all workloads to be part of the mesh for proper functionality.
+====
+
+.Procedure
+
+. Open the application's `Deployment` resource in an editor. In this case, exclude the `ratings-v1` service.
+
+. Modify the `spec.template.metadata.labels` section of your `Deployment` resource to include the label `sidecar.istio.io/inject: false` to disable sidecar injection.
++
+[source,yaml,subs="attributes,verbatim"]
+----
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+name: ratings-v1
```

Comment:
Here we are disabling the injection for the pod `ratings-v1` but in the list of pods shown below Line157, ratings-v1 still has the sidecar. I think you should use the following output at Line 154.

```
NAME                              READY   STATUS    RESTARTS   AGE
details-v1-6bc7b69776-7f6wz       2/2     Running   0          7s
productpage-v1-54f48db985-gd5q9   2/2     Running   0          29m
ratings-v1-5d645c985f-xsw7p       1/1     Running   0          29m
reviews-v1-bd5f54b8c-zns4v        2/2     Running   0          29m
reviews-v2-5d7b9dbf97-wbpjr       2/2     Running   0          29m
reviews-v3-5fccc48c8c-bjktn       2/2     Running   0          29m
```

Reply:
Oops, good catch, perhaps I copy/pasted wrong. I would suggest also updating the "AGE" column as well to make it look more realistic, so that would be:

```
NAME                              READY   STATUS    RESTARTS   AGE
details-v1-6bc7b69776-7f6wz       2/2     Running   0          29m
productpage-v1-54f48db985-gd5q9   2/2     Running   0          29m
ratings-v1-5d645c985f-xsw7p       1/1     Running   0          7s
reviews-v1-bd5f54b8c-zns4v        2/2     Running   0          29m
reviews-v2-5d7b9dbf97-wbpjr       2/2     Running   0          29m
reviews-v3-5fccc48c8c-bjktn       2/2     Running   0          29m
```